app/javascript/mastodon/features/emoji/normalize.ts

```diff
@@ -150,6 +158,21 @@ export function twemojiToUnicodeInfo(
   return hexNumbersToString(mappedCodes);
 }
 
+export function cleanExtraEmojis(extraEmojis?: CustomEmojiMapArg) {
+  if (!extraEmojis) {
+    return null;
+  }
+  if (!isList(extraEmojis)) {
+    return extraEmojis;
+  }
+  return (
+    extraEmojis.toJS() as ApiCustomEmojiJSON[]
```

Comment:
Eugen recently recommended avoiding use of `.toJS()` at runtime. Not sure if it can be avoided here, or if it is performance-critical, just thought I'd mention it

Reply:
swapped to `toJSON`. eventually i want to remove the usage of Immutable entirely from this data structure.